### PR TITLE
Feature discrete combo obj

### PR DESCRIPTION
**Pull request by hlkline**
Allows weighted sum of objectives to be evaluated with the discrete adjoint. (Previously available only with continuous adjoint)
A test case is added for this functionality in serial_regression_AD.py

Also includes:
related clean-up and standardization, such as making sure that a redundancy check inside python functions does the appropriate thing when multiple objectives are used (previously would duplicate some analyses during optimization). 
treating surface heatflux per-surface (making this more similar to how surface forces are treated) so that if multiple surfaces are included only the one specified as being monitored will be used in the objective function. 
removal of some unused variables that turned up during compiler checks.

### CODE REVIEW

**Comment by LaSerpe** [comment]
Hi Heather!
Residuals in regression tests are ok but something goes wrong during the merge of the solution for the parallel fsi2d test case (could it be related to recent changes in SU2_SOL.cpp?)

**Comment by hlkline** [comment]
Thanks for the comment. 
SInce the regression tests all pass, I wonder if this is something that turns up in the current develop branch? This pull request didn't modify the SU2_SOL.cpp file. Can you confirm that the problem does not exist in the develop branch?

**Comment by LaSerpe** [comment]
I just checked and develop has that problem too, I see it from log files in build n 600.2. That's probably something we missed form an older pull request.
Maybe we should start thinking about adding a test to check also those python script (?) just to avoid such issues in the future. Cause right now we are only looking at residuals and we can't check them.

**Comment by hlkline** [comment]
Ok, thanks for checking that. Since it's not directly connected to this pull request, I'm going to open up an issue. Please add whatever observations/additional info you have there to help whoever decides to address it.

File: SU2_CFD/include/solver_structure.hpp [hunk 1]
```diff
@@ -1930,6 +1930,20 @@ class CSolver {
...
+   */
+	virtual su2double GetSurface_HF(unsigned short val_marker);
+
+  /*!
+   * \brief A virtual member.
+   * \param[in] val_marker - Surface marker where the heat flux is computed.
+   * \return Value of the maximum heat flux (viscous contribution) on the surface <i>val_marker</i>.
+   */
+	virtual su2double GetSurface_MaxHF(unsigned short val_marker);
+
```
Comment (arubino) [inline SU2_CFD/include/solver_structure.hpp]: Could you check if the indentation is correct (using 2 spaces)?

File: SU2_CFD/src/solver_direct_mean.cpp [hunk 2]
```diff
@@ -7333,32 +7329,47 @@ void CEulerSolver::Compute_ComboObj(CConfig *config) {
...
+      Total_ComboObj+=obj_weight*Total_CFreeSurface;
       break;
     case AVG_TOTAL_PRESSURE:
+      Total_ComboObj+=obj_weight*OneD_TotalPress;
       break;
     case AVG_OUTLET_PRESSURE:
+      Total_ComboObj+=obj_weight*OneD_PressureRef;
       break;
     case MASS_FLOW_RATE:
+      Total_ComboObj+=obj_weight*OneD_MassFlowRate;
```
Comment (arubino) [inline SU2_CFD/src/solver_direct_mean.cpp]: maybe one could use for `obj_weight` the same style as for the other variables?

File: Common/src/config_structure.cpp [hunk 3]
```diff
@@ -5159,11 +5158,7 @@ string CConfig::GetObjFunc_Extension(string val_filename) {
...
     }
-    else {
-      if (DiscreteAdjoint) {
-        cout << endl << "Combined objective not yet compatible with discrete adjoint. Specify only one OBJECTIVE_FUNCTION." << endl << endl;
-        exit(EXIT_FAILURE);
-      }
+    else{
       AdjExt = "_combo";
     }
     Filename.append(AdjExt);
```
Comment (arubino) [inline Common/src/config_structure.cpp]: This is not related with your changes :) Could you please change the indentation of GetnDim?

File: SU2_CFD/src/solver_direct_mean.cpp [hunk 4]
```diff
@@ -12677,6 +12688,7 @@ CNSSolver::CNSSolver(void) : CEulerSolver() {
   Surface_CL_Visc = NULL; Surface_CD_Visc = NULL; Surface_CSF_Visc = NULL; Surface_CEff_Visc = NULL;
   Surface_CFx_Visc = NULL;   Surface_CFy_Visc = NULL;   Surface_CFz_Visc = NULL;
   Surface_CMx_Visc = NULL;   Surface_CMy_Visc = NULL;   Surface_CMz_Visc = NULL;
+  Surface_HF = NULL; Surface_MaxHF = NULL;
```
Comment (arubino) [inline SU2_CFD/src/solver_direct_mean.cpp]: Are `Surface_HF `and `Surface_MaxHF` used just for a viscous case? If so, maybe adding `_Visc` would help?

**Comment by hlkline** [comment]
I think the last commit addressed your concerns @arubino, thanks for the feedback.
@LaSerpe - that fsi bug was fixed in a separate pull request.

**Comment by hlkline** [comment]
Thanks. 
Waiting for 1 more approval prior to merging.
